### ISSUE TITLE
StackOverflowException is possible when getting the AccessibleName for Combobox.

### ISSUE THREAD

**Issue by SergeySmirnov-Akvelon** [dotnet/winforms#4157]
.NET Core Version: 
- .Net SDK 5.0.100-rc.2.20479.15

**Problem description:**
The "ComboBox.ComboBoxChildDropDownButtonUiaProvider" class uses "get_accNameInternal" method for getting of "Name" property. 
```
internal virtual string? get_accNameInternal(object childID)
{
    if (IsClientObject)
    {
        ValidateChildID(ref childID);

        Debug.WriteLineIf(CompModSwitches.MSAA.TraceInfo, "AccessibleObject.get_accName: this = " + ToString() +
            ", childID = " + childID.ToString());

        // Return the name property if available
        if (childID.Equals(NativeMethods.CHILDID_SELF))
        {
            return Name;
        }

        // If we have an accessible object collection, get the appropriate child
        AccessibleObject? child = GetAccessibleChild(childID);
        if (child != null)
        {
            return child.Name;
        }
    }

    string? retval = systemIAccessible.get_accName(childID);

    if (IsClientObject)
    {
        if (string.IsNullOrEmpty(retval))
        {
            // Name the child after its parent
            retval = Name;
        }
    }

    return retval;
}
```

This method in several branches uses the "Name" property that leads to a new attempt to get "ComboBox.ComboBoxChildDropDownButtonUiaProvider.Name" property and a new call of "get_accNameInternal" method. As a result, we can get a StackOverflowException.

**Comment by RussKie** [comment]
Can this be the root cause of https://github.com/dotnet/winforms/issues/3452?

**Comment by SergeySmirnov-Akvelon** [comment]
No, I don't think that these two bugs are related.

**Comment by RussKie** [comment]
Can you provide a repro, and describe how easy is to hit the issue?

**Comment by SergeySmirnov-Akvelon** [comment]
@RussKie, I found that issue can be reproduced using next unit test:
```
[WinFormsFact]
public void ComboBoxItemAccessibleObject_Name()
{
    using var control = new ComboBox()
    {
        DropDownStyle = ComboBoxStyle.DropDown
    };

    ComboBox.ComboBoxAccessibleObject accessibleObject = Assert.IsType<ComboBox.ComboBoxAccessibleObject>(control.AccessibilityObject);
    var name = accessibleObject.DropDownButtonUiaProvider.Name;
}
```
If the Handle is not created, then we fall into an endless loop when trying to get a name of DropDownButtonUiaProvider.

**Comment by SergeySmirnov-Akvelon** [comment]
Fixed in #4495
